test/services/next_references/next_transaction_reference.service.test.js

```diff
@@ -25,61 +25,110 @@ describe('Next Transaction Reference service', () => {
   })
 
   describe('When a valid region and regime are specified', () => {
-    describe("and the reference is needed for a 'credit note'", () => {
-      it('returns a correctly formatted transaction reference', async () => {
-        const result = await NextTransactionReferenceService.go(regime.id, 'R', 'C')
+    describe("and the ruleset provided is 'presroc'", () => {
+      describe("and the reference is needed for a 'credit note'", () => {
+        it('returns a correctly formatted transaction reference', async () => {
+          const result = await NextTransactionReferenceService.go(regime.id, 'R', 'presroc', 'C')
 
-        expect(result).to.equal('RAC1000001')
+          expect(result).to.equal('RAC1000001')
+        })
       })
-    })
 
-    describe("and the reference is needed for an 'invoice'", () => {
-      it('returns a correctly formatted transaction reference', async () => {
-        const result = await NextTransactionReferenceService.go(regime.id, 'R', 'I')
+      describe("and the reference is needed for an 'invoice'", () => {
+        it('returns a correctly formatted transaction reference', async () => {
+          const result = await NextTransactionReferenceService.go(regime.id, 'R', 'presroc', 'I')
 
-        expect(result).to.equal('RAI1000001')
+          expect(result).to.equal('RAI1000001')
+        })
+      })
+
+      describe('the transaction reference generated', () => {
+        it('increments with each call', async () => {
+          const result = await NextTransactionReferenceService.go(regime.id, 'R', 'presroc', 'I')
+          const secondResult = await NextTransactionReferenceService.go(regime.id, 'R', 'presroc', 'C')
+
+          // The call to slice(-1) grabs the last character from the returned string
```

Comment:
This comment is no longer needed now that `slice(-1)` has been replaced with `endsWith()`